pkg/metrics/tags.go

```diff
@@ -135,6 +135,16 @@ func AugmentWithResponse(baseCtx context.Context, responseCode int) context.Cont
 	return ctx
 }
 
+// AugmentWithResponseAndRouteTag augments the given context with response-code and route-tag specific tags.
+func AugmentWithResponseAndRouteTag(baseCtx context.Context, responseCode int, routeTag string) context.Context {
```

Comment:
Create a func called `AugmentWithRouteTag` then use `AugmentWithRouteTag(AugmentWithResponse(ctx, responsecode), tag)` instead where we need to set both response and tag?

Reply:
Yes, I know what you mean.
Actually, in the beginning, I did it as you commented, but, to avoid to make a new context object for each time to augment the tag label, I merged them. 
WDYT?

Reply:
Currently the Context is constant in QP, since we try to limit the number of allocations on the data path.
Here I presume, we'd have to allocate one on every request, depending on the tag that the request arrived from?

Reply:
Yeah, depending on the tag and the response code.

Reply:
Yes, like the original implementation for the `AugmentWithResponse`.

---
For clarity,
if I understand correctly, `AugmentWith*` functions depend on `go.opencensus.io/tag`.
For each time to call it, it creates new context object and a map,  and then copies a map containing the key and value of the labels.

Reply:
we may want to create an LRU map with contexts, or just absorb the cost.
Don't have a preference here.
/cc @markusthoemmes